src/main/java/emu/grasscutter/game/managers/DeforestationManager/HitTreeRecord.java

```diff
@@ -0,0 +1,77 @@
+package emu.grasscutter.game.managers.DeforestationManager;
+
+
+import java.util.ArrayList;
+
+public class HitTreeRecord {
+    private final static int RECORD_EXPIRED_SECONDS = 60*5; // 5 min
+    private final static int RECORD_MAX_TIMES = 3; // max number of wood
+    private final static int RECORD_MAX_TIMES_OTHER_HIT_TREE = 10; // if hit 10 times other trees, reset wood
+
+    private final int unique;
+    private final ArrayList<Integer> otherTrees = new ArrayList<>(); // hit other tree

```

Comment:
This would probably be better off as a global list that stores the last 10 trees that have been hit, as a circular buffer. It looks like this stores every tree that has been hit quadratically in this current form, which seems like a bad idea.